clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/ArraysTests.scala

```diff
@@ -1192,4 +1193,69 @@ trait ArraysTests { self: BaseIntegrationSpec =>
       storedDocument1.unsafeGet[String]("notes") should be("I'm a note")
     }
   }
+
+  it should "amend documents with new metadata" in {
+
+    val metadataFile: File = File
+      .newTemporaryFile()
+      .overwrite(
+        ArraysMetadata(
+          sampleAlias = Some("amended_sample_alias"),
+          chipType = Some("amended_chip_type")
+        ).asJson.pretty(implicitly)
+      )
+      .deleteOnExit()
+
+    val upsertKey1 = ArraysKey(
+      location = Location.GCP,
+      chipwellBarcode = Symbol(s"barcode$randomId"),
+      version = 1
+    )
+    val upsertKey2 = ArraysKey(
+      location = Location.GCP,
+      chipwellBarcode = Symbol(s"barcode$randomId"),
+      version = 2
+    )
+
+    for {
+      _ <- runUpsertArrays(
+        upsertKey1,
+        ArraysMetadata()
+      )
+      _ <- runUpsertArrays(
+        upsertKey2,
+        ArraysMetadata(
+          chipType = Some("existing_chip_type")
+        )
+      )
+      _ <- runIgnore(
+        ClioCommand.amendArraysName,
+        "--metadata-location",
+        metadataFile.path.toAbsolutePath.toString
+      )
+      amended1 <- runCollectJson(
+        ClioCommand.queryArraysName,
+        "--chipwell-barcode",
+        upsertKey1.chipwellBarcode.name,
+        "--version",
+        upsertKey1.version.toString
+      )
+      amended2 <- runCollectJson(
+        ClioCommand.queryArraysName,
+        "--chipwell-barcode",
+        upsertKey2.chipwellBarcode.name,
+        "--version",
+        upsertKey2.version.toString
+      )
+    } yield {
+
+      val storedDocument1 = amended1.headOption.getOrElse(fail)
+      storedDocument1.unsafeGet[String]("chip_type") should be("amended_chip_type")
+      storedDocument1.unsafeGet[String]("sample_alias") should be("amended_sample_alias")
+
+      val storedDocument2 = amended2.headOption.getOrElse(fail)
+      storedDocument2.unsafeGet[String]("chip_type") should be("existing_chip_type")
+      storedDocument2.unsafeGet[String]("sample_alias") should be("amended_sample_alias")
+    }
+  }
```

Comment:
Should also have a test using the `queryInput` options.

Reply:
Obsolete because there is not more `queryInput` argument.